File: bookstore-service-springboot/src/main/java/com/keabyte/bookstore/api/model/Book.java
package com.keabyte.bookstore.api.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Book {

    private Long id;

    private String title;

    private String author;

    private String description;

    private LocalDate releaseDate;

    private String coverImageUrl;
}

File: bookstore-service-micronaut-kotlin/src/main/kotlin/com/example/micronaut/bookstore/controller/model/Book.kt
package com.example.micronaut.bookstore.controller.model

import com.example.micronaut.bookstore.repository.entity.BookEntity
import io.micronaut.serde.annotation.Serdeable
import java.time.LocalDate

@Serdeable
data class Book(
    val id: String,
    val title: String,
    val author: String,
    val description: String?,
    val releaseDate: LocalDate,
    val coverageImageUrl: String?
) {
    companion object {
        fun from(source: BookEntity): Book {
            return Book(
                id = source.id,
                title = source.title,
                author = source.author,
                releaseDate = source.releaseDate,
                description = source.description,
                coverageImageUrl = source.coverageImageUrl
            )
        }
    }
}